libcfnet/tls_generic.c

```diff
@@ -724,7 +724,20 @@ int TLSRecv(SSL *ssl, char *buffer, int toget)
     assert(toget < CF_BUFSIZE);
     assert_SSLIsBlocking(ssl);
 
-    int received = SSL_read(ssl, buffer, toget);
+    int received,tries = 0,code=0;
+    do
+      {
+        received = SSL_read(ssl, buffer, toget);
```

Comment:
would SSL_peek() be a better choice here?

Reply:
I think we can do `SSL_read()` just fine.